modules/apps/commerce/commerce-product-definitions-web/src/main/resources/META-INF/resources/edit_cp_definition_visibility.jsp

```diff
@@ -69,16 +71,36 @@ Map<String, String> contextParams = HashMapBuilder.<String, String>put(
 <liferay-frontend:component
 	context='<%=
 		HashMapBuilder.<String, Object>put(
-			"accountGroupDataSetId", CommerceProductFDSNames.PRODUCT_ACCOUNT_GROUPS
-		).put(
 			"accountGroupItemSelectorURL", cpDefinitionsDisplayContext.getAccountGroupItemSelectorUrl()
 		).put(
 			"channelDataSetId", CommerceProductFDSNames.PRODUCT_CHANNELS
 		).put(
 			"channelItemSelectorURL", cpDefinitionsDisplayContext.getChannelItemSelectorUrl()
+		).put(
+			"checkedAccountGroupIds", cpDefinitionsDisplayContext.getCheckedCommerceAccountGroupIds()
+		).put(
+			"checkedCommerceChannelIds", cpDefinitionsDisplayContext.getCheckedCommerceChannelIds()
 		).put(
 			"productId", cpDefinition.getCProductId()
 		).build()
 	%>'
 	module="{editCpDefinitionVisibility} from commerce-product-definitions-web"
-/>
\ No newline at end of file
+/>
+
+<aui:script>
+	sessionKey = 'com.liferay.commerce.product.definitions.web.successMessage';
+
+	successMessage = Liferay.Util.SessionStorage.getItem(
+		sessionKey,
+		Liferay.Util.SessionStorage.TYPES.NECESSARY
+	);
+
+	if (successMessage) {
+		Liferay.Util.openToast({
+			message: successMessage,
+			type: 'success',
+		});
+
+		Liferay.Util.SessionStorage.removeItem(sessionKey);
+	}
+</aui:script>
```

Comment:
Personally I don't like this for multiple reasons:
1) missing const in front of sessionKey and successMessage
2) add <aui:script in the jsp
3) the logic (I saw there is another example like this in `modules/apps/change-tracking/change-tracking-web/src/main/resources/META-INF/resources/publications/view_publications.jsp`)

Before a question: why do you need to do a refresh of the page? Is not firing the update dataset event enough?

@gianmarcobrunialti do you have a suggestion on how to handle this? A refresh of the page following by a toast

Reply:
The page refresh is done so that the select an account group button url can be regenerated with the new checkedAccountGroupIds. Without it the checkedAccountGroupIds from the previous action will not show correctly. The dataset update event only updates the table but not the action controls outside of it.

Reply:
[Screencast from 09-05-2024 03:25:35 PM.webm](https://github.com/user-attachments/assets/8a801273-8b4a-4e09-9600-528389772372)